cpp/src/parquet/encoding.cc

```diff
@@ -748,11 +769,11 @@ class PlainByteArrayDecoder : public PlainDecoder<ByteArrayType>,
         data += increment;
         data_size -= increment;
         bytes_decoded += increment;
-        ++i;
       } else {
         ARROW_RETURN_NOT_OK(out->AppendNull());
       }
       bit_reader.Next();
+      ++i;
```

Comment:
Does this fix the bug that had been reported elsewhere? Is this fix covered by a new test added here?

Reply:
Yes, and yes, from an earlier [commit](https://github.com/apache/arrow/pull/3721#issuecomment-467493421):

> The latest commit adds a test case for PARQUET-1537. Just needed to have a null in the input data to reproduce the reported issue.